gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/Context.java

```diff
@@ -152,6 +153,34 @@ public GremlinExecutor getGremlinExecutor() {
         return gremlinExecutor;
     }
 
+    /**
+     * Gets whether the server has started processing the response for this request.
+     */
+    public boolean getStartedResponse() { return startedResponse.get(); }
+
+    /**
+     * Signal that the server has started processing the response.
+     */
+    public void setStartedResponse() { startedResponse.set(true); }
+
+    /**
+     * Writes a default timeout error response message to the underlying channel.
+     */
+    public void sendTimeoutResponse() {
+        sendTimeoutResponse(String.format("A timeout occurred during traversal evaluation of [%s] - consider increasing the limit given to evaluationTimeout", requestMessage));
+    }
+
+    /**
+     * Writes a specific timeout error response message to the underlying channel.
+     */
+    public void sendTimeoutResponse(String message) {
```

Comment:
Nit: All arguments and variables should declared as final unless needed. 
```suggestion
    public void sendTimeoutResponse(final String message) {
```